docs/modules/user-guide/partials/assembly_devfile-library.adoc

```diff
@@ -0,0 +1,31 @@
+ifdef::context[:parent-context-of-assembly_devfile-library: {context}]
+
+
+ifndef::context[]
+[id="assembly_devfile-library"]
+endif::[]
+ifdef::context[]
+[id="assembly_devfile-library_{context}"]
+endif::[]
+= Devfile library
+
+:context: assembly_devfile-library
+
+
+[role="_abstract"]
+Using the devfile library, you can:
+
+* Validate a devfile.
+* Parse a `devfile.yaml` file as specified by the devfile API reference.
+* Write updated data to a `devfile.yaml` file.
+* Generate Kubernetes objects for various devfile resources.
+* Define the `util` functions for the devfile.
+
+The library is written in Golang, and the following projects consume the library as a Golang dependency:
+
+* link:https://github.com/openshift/odo[Odo]
+* link:https://github.com/openshift/console[OpenShift Console]
+
```

Comment:
An assembly file includes modules. See https://redhat-documentation.github.io/modular-docs/#assembly-definition

Reply:
https://raw.githubusercontent.com/redhat-documentation/modular-docs/master/modular-docs-manual/files/TEMPLATE_ASSEMBLY_a-collection-of-modules.adoc

Reply:
Although I can't find it documented anywhere, we're prohibited from linking to GitHub doing so doesn't comply with Red Hat's observance of restrictions on exports to proscribed countries. I just got dinged on this issue in another PR, here: https://github.com/openshift/openshift-docs/pull/37369#discussion_r743602768

Reply:
@jc-berger @bburt-rh The only supporting statement I can find for this "ban" is 
"Links to GitHub need to be okayed by relevant support team rep (for OpenShift, that would be Eric Rich)." in the OpenShift Docs Peer Review Checklist: https://docs.google.com/document/d/1WN7k72PKxcPA__erp5TK-CefmS_sVwE3r40PvpmNk3E/edit?usp=sharing

Reply:
@rolfedh I removed the hyperlinks all together. Better?

Reply:
The "no links to GitHub" rule would apply to all the links in this PR, possibly also the entire doc set. It seems like applying this rule would render the docs unusable because the workflow depends on those links. I leave it to you and your product team to determine how to address this issue. All I can do say is that the practice raises compliance issues that the organization takes seriously.